plugin-server/src/main/ingestion-queues/batch-processing/each-batch-webhooks.ts

```diff
@@ -145,14 +149,14 @@ export async function eachMessageWebhooksHandlers(
         // exit early if no webhooks nor resthooks
         return
     }
-    const event = convertToIngestionEvent(clickHouseEvent)
-
-    // TODO: previously onEvent and Webhooks were executed in the same process,
-    // and onEvent would call convertToProcessedPluginEvent, which ends up
-    // mutating the `event` that is passed in. To ensure that we have the same
-    // behaviour we run this here, but we should probably refactor this to
-    // ensure that we don't mutate the event.
-    convertToProcessedPluginEvent(event)
+    const event = convertToPostIngestionEvent(clickHouseEvent)
+    mutatePostIngestionEventWithElementsList(event)
+
+    // NOTE: This was previously done via some side effect mutations. as it is possible that the webhooks sent are depending on the
+    // format, we do it here again for legacy support.
+    event.elementsList = convertDatabaseElementsToRawElements(event.elementsList ?? [])
+
+    // TODO: What do? The elements list isn't built here anymore where it _used_ to be
```

Comment:
Sorry I wasn't clearer in my comments. Your change here does make us consistent with the past, however it comes with the downside of doing elementsList creation when it's not needed, which we want to avoid.

There are two places, where we use elements:
1. onEvent plugins - in that case the `convertToProcessedPluginEvent` made sure we run `convertDatabaseElementsToRawElements`
2. webhooks - this was here before via `convertToProcessedPluginEvent` call that you removed.

So that means whenever we use elements, we should use them after having ran `mutatePostIngestionEventWithElementsList` and `convertDatabaseElementsToRawElements`, so to make things clearer and easier we should just move `convertDatabaseElementsToRawElements` call into `mutatePostIngestionEventWithElementsList`.

Note that the PR was already handling `mutatePostIngestionEventWithElementsList` being called only where it's really needed.

Reply:
I still don't understand sorry.

This is the webhooks path. So the only thing that will be able to access these properties is the zapier webhooks which are sent the full payload. _If_ we decide it should have elements, then it will always need to be parsed. If we decide it doesn't then we can skip this stuff alltogether?

Reply:
Ah seems like I forgot a third use case Zapier, which is rare, so there are 3 where we need the elementsList and in all of those 3 cases it should also pass through `convertDatabaseElementsToRawElements`.

So let me try to explain what we'll want:
1. onEvent plugins - currently the `convertDatabaseElementsToRawElements` happens within `convertToProcessedPluginEvent`, we can remove it from there and move it into `mutatePostIngestionEventWithElementsList`, because if elements are undefined `convertDatabaseElementsToRawElements` would do nothing
2. webhook filters - currently we call `mutatePostIngestionEventWithElementsList` and `convertDatabaseElementsToRawElements`, but we know that it's not always need, you already call `getElementsList` in the places it's needed, the only thing we're missing is that it doesn't do `convertDatabaseElementsToRawElements`, so again we can just move `convertDatabaseElementsToRawElements` into `mutatePostIngestionEventWithElementsList`
3. Zapier hooks - if it's a zapier hook we should also call `mutatePostIngestionEventWithElementsList` and  `convertDatabaseElementsToRawElements` on the event before, so if we moved `convertDatabaseElementsToRawElements` into `mutatePostIngestionEventWithElementsList` it's clearer. 

Why ^ is better:
1. Calling  `mutatePostIngestionEventWithElementsList` here means we're not doing lazy elements list creation, so all later for `getElementsList` are uselessly lazy as the Elements always exist. This also means we're creating it, when not needed.
2. `convertDatabaseElementsToRawElements` doesn't need to exist and makes it more confusing to understand and read the code, also opens us up to forgetting to do `convertDatabaseElementsToRawElements` when we discover another place we need to do `mutatePostIngestionEventWithElementsList`

Reply:
Alright. I merged it and moved the check way lower down.